cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -347,7 +359,7 @@ runQueryPoolParams (AnyConsensusModeParams cModeParams) network poolid = do
     & hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch (AnyConsensusMode cMode) anyE)
 
   let qInMode = QueryInEra eInMode . QueryInShelleyBasedEra sbe $ QueryDebugLedgerState
-  result <- executeQuery era cModeParams localNodeConnInfo qInMode
+  result <- executeQuery era cModeParams NodeToClientV_1 localNodeConnInfo qInMode
```

Comment:
So this is the type of situation we don't want. If we can implement a function that takes `qInMode` and returns a minimum required node version, that would be ideal.

Reply:
Same as before.